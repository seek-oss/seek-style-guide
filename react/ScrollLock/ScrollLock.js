import React, { Component } from 'react';
import PropTypes from 'prop-types';

import ReactScrollLock from 'react-scrolllock';

export default class ScrollLock extends Component {

  static propTypes = {
    shouldLockBody: PropTypes.bool,
    children: PropTypes.oneOfType([
      PropTypes.arrayOf(PropTypes.node),
      PropTypes.node
    ])
  };

  static defaultProps = {
    shouldLockBody: false,
    children: null
  };

  storeContainerReference = container => {
    if (container !== null) {
      this.container = container;
    }
  }

  render() {
    const { shouldLockBody, children } = this.props;

    return (
      <div ref={this.storeContainerReference}>
        {children}
        {shouldLockBody ? <ReactScrollLock scrollTarget={this.container} /> : null }
      </div>
    );
  }

}
