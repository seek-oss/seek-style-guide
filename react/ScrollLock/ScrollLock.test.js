import { shallow } from 'enzyme';
import React from 'react';

import ScrollLock from './ScrollLock';

describe('ScrollLock', () => {
  it('should render without props', () => {
    const wrapper = shallow(<ScrollLock />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with children', () => {
    const wrapper = shallow(<ScrollLock children={<p>Children</p>} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with body lock', () => {
    const wrapper = shallow(<ScrollLock children="Children" shouldLockBody />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not touch-scroll within children', () => {
    const wrapper = shallow(<ScrollLock children="Children" shouldLockBody shouldScrollInside={false} />);
    expect(wrapper).toMatchSnapshot();
  });
});
