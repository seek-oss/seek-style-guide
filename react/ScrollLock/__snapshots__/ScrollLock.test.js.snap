// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollLock should render with body lock 1`] = `
<div>
  Children
  <Component />
</div>
`;

exports[`ScrollLock should render with children 1`] = `
<div>
  <p>
    Children
  </p>
</div>
`;

exports[`ScrollLock should render without props 1`] = `<div />`;
